### ISSUE TITLE
noUnusedLocals incorrectly flags "non locals"

### ISSUE THREAD

**Issue by wiredprairie** [Microsoft/TypeScript#21677]
`noUsedLocals`  considers class members and properties as locals. When this compiler flag is `true`, the code below is an error. This flag should not consider class members or properties as it's specifically called `noUnsedLocals`. This change has incorrectly flagged many cases in our very large code base that had been ignored in earlier versions of the compiler. 

If this member and property functionality is desirable (which I agree it is), a new compiler flag should be added rather than changing the meaning of already used flags. `noUsedMembers` and `noUnusedProperties` for example. 

Search Terms: `noUnusedLocals`

With this code:

```ts
class JustPrivate {
    private ex1: string;
    private get ex2(): string {
        return this.ex1;
    }
}
```

```ts
D:\temp\noUnusedLocalsIssue>.\node_modules\.bin\tsc --version
Version 2.8.0-dev.20180206

D:\temp\noUnusedLocalsIssue>.\node_modules\.bin\tsc
test.ts(4,17): error TS6133: 'ex2' is declared but its value is never read.
```

**Expected behavior:**
No errors. 

As our code base grows, changing these flags meanings can have a huge impact and leave us with few good choices (spend lots of time changing older code or disabling a flag).

**Actual behavior:**
```test.ts(4,17): error TS6133: 'ex2' is declared but its value is never read.```

**Playground Link:** None, as `noUnusedLocals` is not exposed in the UI.

**Comment by andy-ms** [comment]
That code was an error at least as far back as `typescript@2.0`.

**Comment by wiredprairie** [comment]
Yes -- that was a misunderstanding internally with me about the timing/version and what was affected. I heard "properties" and wrote an example with a member. 😞 

I've edited the issue to reflect the actual unexpected change in behavior.

**Comment by RyanCavanaugh** [comment]
> This flag should not consider class members or properties

The intuition is that `private` members are "local" to the class because there's no legal access to those identifiers outside the class body. Unused `private` class methods were also flagged under this rule since its creation; *only* `private` getters/setters were (incorrectly) unchecked, which I think is obviously a bug.

Our general policy is not to add a commandline flag to enable pre-bug behavior whenever there's a bug, even if that bug creates a compat difference. For example, it used to be the case that you could write `f(x: any): x is Foo<Bar>` even if `Bar` didn't meet the generic constraint of `Foo`, but there is no corresponding `--disallowIllegalInstantiationInTypePredicates` flag that you have to opt in to.

You could make `ex2` `protected` instead if the intent is to access it from outside the class. Alternatively if you want the unused checking, the TS Lint rule `no-unused-variable` is a bit more configurable.
